Multisig/Logic/Models/Safe.swift

```diff
@@ -15,6 +15,40 @@ extension Safe: Identifiable {
         self.createdAt = Date()
     }
 
+    func save() {
+        CoreDataStack.shared.saveContext()
+    }
+
+    static func download(at address: String) throws {
+        _ = try App.shared.safeRelayService.safeInfo(at: address)
+    }
+
+    static func alreadyExists(_ address: String) throws -> Bool {
+        let context = CoreDataStack.shared.persistentContainer.viewContext
+        let count = try context.count(for: Safe.by(address: address))
+        return count > 0
+    }
+
+    static func create(address: String, name: String, selected: Bool = true) {
+        let context = CoreDataStack.shared.persistentContainer.viewContext
+
+        let safe = Safe(context: context)
+        safe.address = address
+        safe.name = name
+
+        if selected {
+            let settings = AppSettings.getOrCreate(context: context)
+            settings.selectedSafe = address
+        }
+
+        do {
```

Comment:
please use CoreDataStack.shared.saveContext()

Reply:
becasue I am dealing with an explicit context here, I would just save it, not using implicit one in another function